webapp/src/components/github_repo_selector/github_repo_selector.jsx

```diff
@@ -30,25 +33,69 @@ export default class GithubRepoSelector extends PureComponent {
     }
 
     componentDidMount() {
-        this.props.actions.getRepos();
+        this.props.actions.getOrgs(true);
+        if (this.props.yourOrgs.length > 0) {
+            const firstOrg = this.props.yourOrgs[0];
+            this.props.actions.getReposByOrg(firstOrg.isLoggedInUser ? '' : firstOrg.login);
```

Comment:
handle race condition, if org from cofig settings provided

Reply:
What is the race condition to handle?

Reply:
It was not exactly a race condition; rather, it was a bug. Let's consider a scenario where a GitHub organization is configured in the config.json file. In this case, only one organization will be returned. Within the component, if the length of the organizations is equal to 1, we don't show the react-select for organizations. The expectation is that all the repositories under that organization should have already been fetched for the repository react-select.

However, if we look closely, we observe that the organizations are fetched in the `componentDidMount` method, and there is no opportunity to fetch repositories for the configured organization elsewhere. Additionally, since we are not displaying the organization react-select, the user cannot choose an organization.

Solution: Modified the code by adding an `if `condition in the `componentDidUpdate` method. If `prevProps.yourOrgs` is not equal to `currProps.yourOrgs`, automatically set the first organization and fetch the repositories for that organization.

Reply:
We could alternatively provide the `GithubOrg` config setting to the client so it has some more context on what should be shown